compaction/compaction_manager.hh

```diff
@@ -362,6 +358,26 @@ public:
     // An inline constructor for testing
     compaction_manager(for_testing_tag) : compaction_manager() {}
 
+    const auto& compaction_sg() const noexcept {
+        return _cfg.compaction_sched_group;
+    }
+
+    const auto& maintenance_sg() const noexcept {
+        return _cfg.maintenance_sched_group;
+    }
+
+    auto available_memory() const noexcept {
+        return _cfg.available_memory;
+    }
+
+    auto static_shares() const noexcept {
+        return _cfg.static_shares.get();
+    }
+
+    auto throughput_mbs() const noexcept {
```

Comment:
minor: it's better to restrict automatic return type for complex types, for example:

```
template <typename Func, typename T>
auto future_invoke(Func&& func, T&& v) {
    if constexpr (std::is_same_v<T, monostate>) {
        return std::invoke(std::forward<Func>(func));
    } else {
        return std::invoke(std::forward<Func>(func), std::forward<T>(v));
    }
}
```
the return type is of great help to document a function, the user knows what's expected, so better use it whenever possible.

Reply:
Agree, we should avoid using auto in return and parameter types just for convenience.

Reply:
The main problem with that is that if the member type changes and the get method may implicitly convert it to the old type with no warning, while `auto` convey exactly what we want to do - i.e. return the value or a reference to it, whatever type it is.

Reply:
> The main problem with that is that if the member type changes and the get method may implicitly convert it to the old type with no warning, while `auto` convey exactly what we want to do - i.e. return the value or a reference to it, whatever type it is.

That same problem can still happen to the users of the interface. I think auto as return type should be restricted for complex types where template is involved.

Otherwise every class out there having an integer as member could also use auto when exposing it through its public interface. The return type is very important for documenting the interface too.

If you change a interface from signed to unsigned, then you must be careful that every user of the interface won't break.

Reply:
Done in v6